aten/src/ATen/core/jit_type.h

```diff
@@ -1319,12 +1319,25 @@ struct CAFFE2_API ClassType : public Type {
     // same can subtype from each other.
     return *this == *rhs;
   }
+
   std::string str() const override {
-    return std::string("ClassType<") + typename_ + ">";
+    return std::string("ClassType<") + basename_ + ">";
   }
 
   std::string python_str() const override {
-    return typename_;
+    return qualifiedName_;
+  }
+
+  const std::string& qualname() const {
```

Comment:
We have a QualifiedName object in python_print.cpp that should be used here. Otherwise, expect endless bugs about passing an unqualified name when a qualified one was expected and vice-versa. It also will keep name parsing code in the parser.